### PR TITLE
Add option for detecting strings when tokenizing a program

### LINKED ISSUE #9: Option for string tokenization
When tokenizing a program, tokens inside of strings are replaced with single-character tokens rather than the longest token available. This is not always desirable, as strings are sometimes stored into equations or lists or evaluated. There should be an option in TH_Tokenized::makeDataFromString to tokenize the contents of strings in the same way as the rest of the program.

### PR DESCRIPTION
**Pull request by commandz0**
Fix https://github.com/adriweb/tivars_lib_cpp/issues/9

### CODE REVIEW

File: src/TypeHandlers/TH_Tokenized.cpp [hunk 1]
```diff
@@ -34,9 +34,15 @@ namespace tivars
...
-        (void)options;
         data_t data;
 
+        const auto& optIter = options.find("_detect_strings");
+        bool detect_strings = true;
+        if (optIter != options.end())
+        {
+            detect_strings = optIter->second;
+        }
+
```
Comment (adriweb) [inline src/TypeHandlers/TH_Tokenized.cpp]: Nah, you can use something like:

```suggestion
        const bool detect_strings = !has_option(options, "detect_strings") || options.at("detect_strings") != 0;
```

File: src/TypeHandlers/TH_Tokenized.cpp [hunk 2]
```diff
@@ -49,7 +55,10 @@ namespace tivars
             const std::string currChar = str.substr(strCursorPos, 1);
             if (currChar == "\"")
             {
-                isWithinString = !isWithinString;
+                if(detect_strings)
+                {
+                    isWithinString = !isWithinString;
+                }
```
Comment (adriweb) [inline src/TypeHandlers/TH_Tokenized.cpp]: Can you put `if (detect_strings)` around the whole if/else lines 56->65?

**Review by adriweb — CHANGES_REQUESTED** [review]

**Comment by adriweb** [comment]
(feel free to force-push the changes to your branch, I prefer merging just one commit)

**Comment by adriweb** [comment]
Oh also could you add a test for that in tests.cpp?

File: src/TypeHandlers/TH_Tokenized.cpp (hunk 1 above)
Comment (commandz0) [reply src/TypeHandlers/TH_Tokenized.cpp]: Ah, neat. I'm not super familiar with C++, so I just copied how the generic reals did it.

Comment (adriweb) [reply src/TypeHandlers/TH_Tokenized.cpp]: Yeah, I should also use `has_option` there :D Also, this is just a function defined in utils.

**Review by adriweb — APPROVED** [review]

**Pull request merged**